compiler/aarch64/codegen/ControlFlowEvaluator.cpp

```diff
@@ -466,11 +471,173 @@ OMR::ARM64::TreeEvaluator::acmpeqEvaluator(TR::Node *node, TR::CodeGenerator *cg
 	return OMR::ARM64::TreeEvaluator::unImpOpEvaluator(node, cg);
 	}
 
+static uint32_t sumOf2ConsecutivePowersOf2(uint32_t numberOfCases)
+   {
+   for (uint32_t i = 3; i < 0xc0000000; i <<= 1)
+      {
+      if (i == numberOfCases)
+         {
+         return ((i & (i - 1)) >> 1) + 1;
+         }
+      }
+   return 0;
+   }
+
+static void binarySearchCaseSpace(TR::CodeGenerator *cg, TR::Register *selectorReg, TR::Node *lookupNode, uint32_t lowChild, uint32_t highChild, bool &evaluateDefaultGlRegDeps)
+   {
+   uint32_t numCases = highChild - lowChild + 1;
+   uint32_t pivot;
+   if (0 == (pivot = sumOf2ConsecutivePowersOf2(numCases)))
+      {
+      pivot = lowChild + (numCases / 2) - 1;
+      }
+   else
+      {
+      pivot += lowChild - 1;
+      }
+
+   if (pivot >= lowChild)
+      {
+      int32_t pivotValue = lookupNode->getChild(pivot)->getCaseConstant();
+      generateCompareImmInstruction(cg, lookupNode, selectorReg, pivotValue);
```

Comment:
`generateCompareImmInstruction()` can take immediate values between 0 and 4095 inclusive (`constantIsUnsignedImm12()`) only.

Reply:
I opened #4677 for this.